app/js/components/validation.js

```diff
@@ -85,8 +85,30 @@ if (form.length) {
     });
 }
 
-// Add a stricter url validator, this one ensures the protocol is set on the input value
+// Add a stricter url validator, fields validated with urlstrict are
+// allowed to be empty, or a URL with protocol.
 window.Parsley.addValidator('urlstrict', function (value, requirement) {
+    return validateEmpty(value) || validateUrl(value);
+}, 32).addMessage('en', 'urlstrict', 'This value should be a valid URL.');
+
+// Add URI validator (must be URN or URL), fields validated with urn
+// must be empty or be a valid URN  or URL (with protocol).
+window.Parsley.addValidator('uri', function (value, requirement) {
+    return validateEmpty(value) || validateUrl(value) || validateUrn(value);
+}, 32).addMessage('en', 'uri', 'This value should be a valid URI.');
```

Comment:
My opinion is that using URI in this context is possibly confusing as you are not allowed to fill any URI but just URL's and URN's. 

For example, I'm not allowed to fill this valid URI into the Entity ID field: `mailto:someone@example.com`

Reply:
My suggestion is to substitute uri with urn.